data/simpy.gram

```diff
@@ -211,17 +211,23 @@ class_def: [decorators] 'class' NAME ['(' [arguments] ')'] ':' block
 
 block[asdl_seq*]: simple_stmt | NEWLINE INDENT a=statements DEDENT { a }
 
-star_full_expressions: ','.star_full_expression+ [',']
+expressions_list[asdl_seq*]: a=','.star_expression+ [','] { a }
 expressions[expr_ty]:
-    | '*' bitwise_or (',' ('*' bitwise_or | expression))+ [',']
-    | '*' bitwise_or ','
-    | '*' bitwise_or
-    | expression (',' ('*' bitwise_or | expression))+ [',']
-    | expression ','
+    | a=star_expression b=(',' c=star_expression { c })+ [','] {
+        _Py_Tuple(seq_insert_in_front(p, a, b), Load, EXTRA_EXPR(a, seq_get_tail(NULL, b))) }
+    | a=star_expression ',' { _Py_Tuple(singleton_seq(p, a), Load, EXTRA_EXPR(a, a)) }
+    | star_expression
+star_expression[expr_ty]:
+    | a='*' b=bitwise_or { _Py_Starred(b, Load, EXTRA(a, token_type, b, expr_type)) }
     | expression
 
-star_full_expression: '*' bitwise_or | full_expression
-full_expression: NAME ':=' expression | expression
+star_full_expressions[asdl_seq*]: a=','.star_full_expression+ [','] { a }
```

Comment:
I think we should add some explicit cases for these rules, because the pattern `"SOMETHING".rulename "SOMETHING"` makes me now suspect (not that this is wrong, but this was failing very subtly on the assignment PR) of the parser eagerly consuming the "SOMETHING".

Reply:
Added a new test file `list_trailing_comma.py` that uses `star_named_expressions` and a trailing comma. I don't think there is a problem. `','.something+` is designed to be the same with `something (',' something)*`, which is what `star_named_expressions` was before. In the assignment case, the rule was actually somewhat different, which is why we missed it at the beginning.

Reply:
Yup, there is no problem but I still think having a test case for this is slightly better in case we do future refactors or similar. Maybe is just me fearing the wrath of PEG with these patterns :)